iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/stack.inl

```diff
@@ -30,8 +30,9 @@ inline cxx::optional<T> stack<T, Capacity>::pop() noexcept
         return cxx::nullopt;
     }
 
-    // low level memory management with access to the topmost element on the untyped buffer;
-    // type safety is ensured by the template parameter
+    // AXIVION Next Construct CertC++-EXP36 : every entry of m_data is aligned to alignof(T)
```

Comment:
The way it is here only the first element is aligned. By aligning the element type this can be fixed and may make the suppression unnecessary.

The way it currently is, is a bug for certain types sizes and alignments.